admin/services/administration-message.service.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+const administrationMessageDataService = require('./data-access/administration-message.data.service')
+const redisCacheService = require('./redis-cache.service')
+const emptyFieldsValidator = require('../lib/validator/common/empty-fields-validators')
+const serviceMessageErrorMessages = require('../lib/errors/service-message')
+
+const administrationMessageService = {}
+const serviceMessageRedisKey = 'serviceMessage'
+
+/**
+ * Get the current service message
+ * @returns {object}
+ */
+administrationMessageService.getMessage = async () => {
+  let cachedServiceMessage
+  const result = await redisCacheService.get(serviceMessageRedisKey)
+  try {
+    cachedServiceMessage = JSON.parse(result)
+    if (cachedServiceMessage) {
+      return cachedServiceMessage
+    }
+  } catch (ignore) {}
+  return administrationMessageDataService.sqlFindActiveServiceMessage()
+}
+
+/**
+ * Validates and stores the service message data
+ * @param {object} requestData
+ * @param {string} userId
+ * @returns {Promise<*>}
+ */
+administrationMessageService.setMessage = async (requestData, userId) => {
+  if (!userId) {
+    throw new Error('User id not found in session')
+  }
+  const { serviceMessageTitle, serviceMessageContent } = requestData
+  const serviceMessageErrors = emptyFieldsValidator.validate([
+    { fieldKey: 'serviceMessageTitle', fieldValue: serviceMessageTitle, errorMessage: serviceMessageErrorMessages.emptyServiceMessageTitle },
+    { fieldKey: 'serviceMessageContent', fieldValue: serviceMessageContent, errorMessage: serviceMessageErrorMessages.emptyServiceMessageContent }
+  ])
+  if (serviceMessageErrors.hasError()) {
+    return serviceMessageErrors
+  }
+
+  const serviceMessageData = administrationMessageService.prepareSubmissionData(requestData, userId)
+
+  requestData.isEditView
+    ? await administrationMessageDataService.sqlUpdate(serviceMessageData) : await administrationMessageDataService.sqlCreate(serviceMessageData)
+  return redisCacheService.set(serviceMessageRedisKey, serviceMessageData)
```

Comment:
There appears to be an incompatibility between the redis and SQL implementations.  The SQL DDL allows multiple service messages but the redis cache only allows one.

Reply:
@jon-shipley this has been changed as there's no edit action anymore.